### ISSUE TITLE
Install of boost fails because: not enough values to unpack (expected 2, got 1)

### ISSUE THREAD

**Issue by patlecat** [conan-io/conan#6793]
So I freshly installed Conan 1.24 as an installer on my new Windows10 with MSVC2019 and I ran into this bug that seems to pop up from time to time in the Conan community. But as a beginner with Conan I have no clue at all what to do and I think installing Boost should be the easiest thing to do since it's surely well tested.

Here's my **conanfile.txt** in ../boost/1.72.0:
```
[requires]
boost/1.72.0

[options]
asio_no_deprecated: True
bzip2: True
filesystem_no_deprecated: True
lzma: True
system_no_deprecated: True
zlib: True
zstd: True
```
-------------------------------------
The command output:
```
λ conan install .
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

Traceback (most recent call last):
  File "conan\conans\client\command.py", line 2002, in run
  File "conan\conans\client\command.py", line 495, in install
  File "conan\conans\client\conan_api.py", line 89, in wrapper
  File "conan\conans\client\conan_api.py", line 573, in install
  File "conan\conans\client\manager.py", line 54, in deps_install
  File "conan\conans\client\graph\graph_manager.py", line 112, in load_graph
  File "conan\conans\client\graph\graph_manager.py", line 139, in _load_root_node
  File "conan\conans\client\graph\graph_manager.py", line 176, in _load_root_consumer
  File "conan\conans\client\loader.py", line 215, in load_conanfile_txt
  File "conan\conans\client\loader.py", line 241, in _parse_conan_txt
  File "conan\conans\model\options.py", line 282, in loads
  File "conan\conans\model\options.py", line 184, in __init__
ValueError: not enough values to unpack (expected 2, got 1)

ERROR: not enough values to unpack (expected 2, got 1)
```

**Comment by memsharded** [comment]
Hi @patlecat 

Please check the syntax of options definition in https://docs.conan.io/en/latest/using_packages/conanfile_txt.html#options

The syntax is with "=", as the ":" is intended to define the package, you might need to specify the options as ``boost:option=value``

**Comment by memsharded** [comment]
This might be an easy to fix UX improvement, that outputs a more clear message clarifying about the input syntax.

**Comment by patlecat** [comment]
Ah yes you're right the syntax was wrong. But now I got these errors below and I do not understand why canon wants to do the downloading and building in it's own folder instead of the one I started it from and contains the conanfile.txt?

--------

```
WARN: patches/bcp_namespace_issues_1_72.patch: file 1/1:         b'Jamroot'                                                              
WARN: patches/bcp_namespace_issues_1_72.patch:  hunk no.1 doesn't match source file at line 143                                          
WARN: patches/bcp_namespace_issues_1_72.patch:   expected: b'import tools/boost_install/boost-install ;'                                 
WARN: patches/bcp_namespace_issues_1_72.patch:   actual  : b'# Backslash because of `bcp --namespace`'                                   
WARN: patches/bcp_namespace_issues_1_72.patch: file 1/1:         b'Jamroot'                                                              
WARN: patches/bcp_namespace_issues_1_72.patch:  hunk no.2 doesn't match source file at line 311                                          
WARN: patches/bcp_namespace_issues_1_72.patch:   expected: b'# stage and install targets via boost-install, above.'                      
WARN: patches/bcp_namespace_issues_1_72.patch:   actual  : b'# Creates a library target, adding autolink support and also creates'       
WARN: patches/bcp_namespace_issues_1_72.patch: already patched  b'Jamroot'                                                               
WARN: patches/boost_core_qnx_cxx_provide___cxa_get_globals.patch: file 1/1:      b'boost/core/uncaught_exceptions.hpp'                   
WARN: patches/boost_core_qnx_cxx_provide___cxa_get_globals.patch:  hunk no.1 doesn't match source file at line 59                        
WARN: patches/boost_core_qnx_cxx_provide___cxa_get_globals.patch:   expected: b'#if !defined(__FreeBSD__) && \\'                         
WARN: patches/boost_core_qnx_cxx_provide___cxa_get_globals.patch:   actual  : b'// On QNX SDP 7.0 (QCC 5.4.0), there are multiple cxxabi.
h, one from glibcxx from gcc and another from libc++abi from LLVM. Which one is included will be determined by the qcc'                  
WARN: patches/boost_core_qnx_cxx_provide___cxa_get_globals.patch: already patched  b'boost/core/uncaught_exceptions.hpp'                 
boost/1.72.0: Building your package in C:\.conan\3544dc\1                                                                                
boost/1.72.0: Generator txt created conanbuildinfo.txt                                                                                   
boost/1.72.0: Calling build()                                                                                                            
boost/1.72.0: WARN: Patching user-config.jam                                                                                             
boost/1.72.0:                                                                                                                            
boost/1.72.0: ERROR: Package '769644ba7de79bcd14f9c0e5f80cdf5c06eb9b39' build failed                                                     
boost/1.72.0: WARN: Build folder C:\.conan\3544dc\1                                                                                      
ERROR: boost/1.72.0: Error in build() method, line 405                                                                                   
        self._create_user_config_jam(self._boost_build_dir)                                                                              
while calling '_create_user_config_jam', line 720                                                                                        
        contents += create_library_config("lzma")                                                                                        
while calling 'create_library_config', line 701                                                                                          
        includedir = self.deps_cpp_info[name].include_paths[0].replace('\\', '/')                                                        
        KeyError: 'lzma'
```

**Comment by memsharded** [comment]
Hi @patlecat 

Please read carefully the https://docs.conan.io/en/latest/getting_started.html and the Conan docs in general. Conan uses a local cache for installing (and building from sources if necessary) the dependencies. Boost will not be installed or built in your user folder. Conan will only put there the necessary files for your build system, (that are also temporary, shouldn't be commited to scm). I would encourage to complete the first parts of the Conan documentation, which are tutorial like and can be executed as you read them, and when everything works, start from there.

In your case, it is possible that the Boost recipe doesn't support properly that specific combination of options, and in that case it would deserve an issue in the https://github.com/conan-io/conan-center-index repo, that is the one containing the recipes for ConanCenter. I would suggest to start also first with the default Boost options, be sure that works, and then start from there, until the reason of failure of the boost recipe can be reported in the conan-center-index repo.

**Comment by patlecat** [comment]
@memsharded Hi James
Thanks for the help so far. I read the getting started with the oversimplified and outdated MD5 example but it doesn't cover the most popular lib for C++ unfortunately. That would be a great addition though.
I generally found the docs not that easy to get into and there are no good beginner vids with hands-on tutorials, only very lengthy talks about the high level concepts of Conan.

But I hope I get around all that soon since I want to use it for a rather big OS project.

**Comment by memsharded** [comment]
merged, will be released in Conan 1.25
